core/aggregation.go

```diff
@@ -303,6 +303,10 @@ func (a *StdSignatureAggregator) ReceiveSignatures(
 		// Verify that the aggregated public key for the quorum matches the on-chain quorum aggregate public key
 		// sans non-signers of the quorum
 		quorumAggKey := state.AggKeys[quorumID]
+		if quorumAggKey == nil {
+			a.Logger.Error("no aggregate public key found for quorum", "quorumID", quorumID)
+			continue
+		}
```

Comment:
Can we just return an error instead of logging one? I'm concerned that we may be skipping validation that might result in the batcher producing a batch that's invalid. Returning an error here should cause the bath to fail without crashing the pod.